prov/efa/test/efa_unit_test_common.c

```diff
@@ -22,23 +67,33 @@ int efa_unit_test_resource_construct(struct efa_resource *resource)
 	if (ret)
 		goto err;
 
-	if (resource->eq_attr) {
-		ret = fi_eq_open(resource->fabric, resource->eq_attr, &resource->eq, NULL);
-		if (ret)
-			goto err;
-	}
+	ret = fi_eq_open(resource->fabric, &eq_attr, &resource->eq, NULL);
+	if (ret)
+		goto err;
+
+	fi_ep_bind(resource->ep, &resource->eq->fid, 0);
 
 	ret = fi_av_open(resource->domain, &av_attr, &resource->av, NULL);
 	if (ret)
 		goto err;
 
+	fi_ep_bind(resource->ep, &resource->av->fid, 0);
+
 	ret = fi_cq_open(resource->domain, &cq_attr, &resource->cq, NULL);
 	if (ret)
 		goto err;
 
+	fi_ep_bind(resource->ep, &resource->cq->fid, FI_SEND | FI_RECV);
+
+	ret = fi_enable(resource->ep);
```

Comment:
`fi_enable` could require additional mocks as we add more tests, e.g. rdma-core.

If a user simply wants to test a component that does not an enabled endpoint, she will still be required to setup the mocks, which can be repetitive.

Same as my comment above, I support the idea of hiding nuanced details for common test cases, but we should do that by composing rather than embedding smaller functions .